### DISCUSSION TITLE
How to start contributing for Robolectric's GSoC 2024?

### DISCUSSION THREAD

**Discussion by Devashishbasu** [robolectric/robolectric#8865] [General]
Hello everyone...can someone please guide me ..how can I start contributing?

**Comment by utzcoz** [comment]
1. Read and learn what Robolectric does: https://github.com/robolectric/robolectric.
2. Search issues with "help wanted" or "start issue" in Robolectric's issue list and try to help some simple issues.
3. Send PRs for existed issues or you found new issues, and maintainer and contributors will help you land the PR if it is meaningful to us.

All ideas of Robolectric for GSoC 2024 requires contributors to know Robolectric before participating in. So it's recommended to write Robolectric tests for Android apps and learn what it can do and what is the limitation of it. 

If you're interested in ideas like supporting Robolectric Native Runtime for Window, it's better you can learn the basic of AOSP development from https://source.android.com/docs/setup/about.

**Reply by Devashishbasu** [reply]
Thank You

**Reply by cjj20** [reply]
@utzcoz So we don't have to come up with new project ideas or features for GSOC?

**Reply by utzcoz** [reply]
@cjj20 Please check https://docs.google.com/presentation/d/1Wyqkef0bK3c9EbZzTdFw7S9Hx5-fq9iRgNbVWozQ2q0 .

**Comment by Devashishbasu** [comment]
Hi @utzcoz  i am getting this error while building the project...could you please help?

The project is using an incompatible version (AGP 8.2.2) of the Android Gradle plugin. Latest supported version is AGP 8.1.0
See Android Studio & AGP compatibility options.

**Reply by MGaetan89** [reply]
@Devashishbasu which version of Android Studio are you using? Can you try with the [latest stable version](https://developer.android.com/studio)?

**Reply by Devashishbasu** [reply]
Thank you so much... I was able to build project with Hedgehog..previously I was using Iguana

**Reply by utzcoz** [reply]
Robolectric tries to use the latest toolchains, including AGP, Gradle, and other dependencies as much as possible. So it's better to use latest stable Android Studio to run latest Robolectric master branch code. And it's recommended to update Android Studio versions automatically to keep the compatibility with latest changes. Anyway, building and tests with `./gradlew` can always work as our CI guards it.

**Reply by Devashishbasu** [reply]
Thank you for the info.

**Comment by Devashishbasu** [comment]
Hello everyone...is there any proposal template?

**Reply by utzcoz** [reply]
Maybe can follow https://google.github.io/gsocguides/student/writing-a-proposal.

**Comment by Devashishbasu** [comment]
Hello everyone!... Are there any project idea documents available? I'm keen to understand the project outcomes, milestones, and expected contributions from contributors.

**Reply by Devashishbasu** [reply]
@MGaetan89 @utzcoz @hoisie

**Reply by MGaetan89** [reply]
Hey @Devashishbasu,
you can check the list of projects here for GSoC 2024 [here](https://docs.google.com/presentation/d/1Wyqkef0bK3c9EbZzTdFw7S9Hx5-fq9iRgNbVWozQ2q0/edit?usp=sharing).

**Comment by Devashishbasu** [comment]
Greetings everyone! After reviewing the project ideas list, I found myself particularly intrigued by two proposals: "Better Activity Contexts" and "Replacing looseSignatures." I'm keen to delve deeper into these projects. Could anyone provide additional details on these initiatives? Furthermore, I'd appreciate insight into what is expected from contributors for these projects.?

**Reply by utzcoz** [reply]
I can explain more for "Replacing looseSignatures". The Google Doc attached as project ideas show some links for issues that we have met when introducing new Android SDK's API shadows to Robolectric, and how we fixed these issues with `looseSignature`. But it's sometimes annoying, and it's not liked by many maintainers because we lost many type information from shadow code. So we want to create a new annotation that can fix the same issues and provide more type information and keep more types as much as possible in our code. For example, I have made some proposals and workable solutions:

https://github.com/robolectric/robolectric/pull/8829
https://github.com/robolectric/robolectric/pull/8841

You can check my PRs to learn what issues that we want to fix.

For this year's GSoC, we hope there is a stable and replacement solution, and we hope its performance is great. And we also hope some important shadow files that use looseSignature can migrate to new annotations. Ideally, we hope we can remove all looseSignature entirely from shadow files, but it is not restrict requirement.

I think @hoisie can help to explain "Better Activity Contexts" better.

**Reply by Devashishbasu** [reply]
Thanks

**Comment by Devashishbasu** [comment]
Hi @utzcoz @MGaetan89 Here is my first version of the proposal: Replacing looseSignatures I would greatly appreciate it if you could spare some time to provide me with some suggestions!
https://docs.google.com/document/d/1Fsqa6zQjgkOa8YV5_3OKl78-H0AJfzW0-N_xX4sVIwA/edit?usp=sharing

**Reply by Devashishbasu** [reply]
Gentle reminder @MGaetan89 @utzcoz @hoisie

**Reply by utzcoz** [reply]
Thanks for your proposal. I will check it ASAP.

**Reply by Devashishbasu** [reply]
Thanks

**Reply by utzcoz** [reply]
It looks great to me. Thanks for your proposal. If the GSoC site opens the proposal submitting, you can submit this version as the first version. I think it can be updated before the deadline.

I appreciate it very much that you can allocate too much time for the project, but I recommend you to reduce your promise in your proposal to get a better life balance if you pass the selection.

**Reply by Devashishbasu** [reply]
Thank you so much for review and suggestion

**Comment by utzcoz** [comment]
@ksachin7  You can post your questions, ideas and thoughts here.

**Reply by ksachin7** [reply]
@utzcoz @hoisie sure, this is my gsoc proposal , looking forward to your review - 
[https://docs.google.com/document/d/1hX8K28XO7ZJfgZfXVZ4BkQkAgME7p_twxfK0hQAHxLk/edit?usp=sharing](url)

**Reply by utzcoz** [reply]
@ksachin7 Please give me and hoisie permission to read this documentation.

**Reply by ksachin7** [reply]
https://docs.google.com/document/d/1hX8K28XO7ZJfgZfXVZ4BkQkAgME7p_twxfK0hQAHxLk/edit?usp=sharing

You can access it through this link @utzcoz

**Reply by utzcoz** [reply]
Please attach more personal information about your education or experiences. It't also recommended to add more links that let us know you have basic development skills. If you don't want to make it public, you can send a modified version to me and hoisie with email. I think you can find our email address from GitHub.

**Reply by utzcoz** [reply]
@ksachin7 If you would, you can pose this proposal to the GSoC system first. I will review this proposal carefully this weekend, and I think it can be updated to the GSoC site before the deadline. I will reply to you with email. Thanks for your proposal.

**Comment by Devashishbasu** [comment]
Hi @utzcoz I am unable to find your email address...can you please provide me I want to discuss something

**Reply by utzcoz** [reply]
You can use my GitHub id as my outlook email id

**Reply by utzcoz** [reply]
Sorry, I will respond email ASAP.

**Reply by Devashishbasu** [reply]
Thanks

**Comment by raehat** [comment]
@utzcoz Hey! I know basics of AOSP development and I find "Native runtime support for Windows" interesting. The summary given in project ideas says "The project would involve fixing bugs and edge cases in Windows support". So are contributors expected to add windows support to RNR or is that process ongoing and contributors are expected to fix bugs?

**Reply by hoisie** [reply]
@raehat Robolectric 4.12 was just released yesterday, and that added support for native SQLite and graphics. However, there are some tricky issues we are facing related to running native code on Windows. For instance, on Windows it is harder to delete temp files that are being used in native code. So the project is mostly fixing issues in Windows.

**Reply by raehat** [reply]
@utzcoz @hoisie any open issues I can look into?

**Comment by harshal-rembhotkar** [comment]
HI @utzcoz ,  How can we submit a draft proposal? Do we have to email it?

**Reply by utzcoz** [reply]
Not. You can share google doc or mail it to us. It's just for previewing and discussion. But before the deadline, you need to submit the proposal to GSoC site.

**Comment by raehat** [comment]
Hey @utzcoz, I mailed you my proposal, if possible please provide some feedback, I would really appreciate it!

**Reply by utzcoz** [reply]
Thanks. Please add @hoisie too. He has more context about this project for native runtime on Windows. I will give my feedback ASAP.

**Reply by raehat** [reply]
Send him too!

**Comment by utzcoz** [comment]
Hello, potential contributors:

1. If you want to participate one project, you can share your thought here. We can discuss first to introduce more context about the project.
2. The GSoC is in the proposal reviewing period, and we recommend you prepare proposal based on discussion and your thought as early as possible, and share with us with messages or emails that you want. We can discuss based on the proposal before the deadline of GSoC proposal submit.
3. We recommend you attach personal information, especially your education experience, your GitHub account and recent contributions when you want to represent yourself skills. If you don't want to keep them secret, you can try to send them to us with email only. And we recommend you attach these information on your final submission on GSoC site for the final reviewing.
4. If you think time is not enough, you can submit your proposal to the GSoC directly.
5. If you want to ask questions, but you don't want it public, please send emails to us.

Looking forward to seeing your proposals and discussing details of these proposals.

**Comment by utzcoz** [comment]
Hello, potential contributors:

> The GSoC 2024 contributor proposal submission period has closed.

Thanks for your interested for Robolectric GSoC 2024, and the contributor proposal submission period has closed. And we will review submitted proposals ASAP.
